src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -168,18 +168,18 @@ internal override string GetDebuggerDisplay()
             SyntaxNode forbiddenZone)
         {
             Debug.Assert(
-                nodeToBind.Kind() == SyntaxKind.CasePatternSwitchLabel ||
-                nodeToBind.Kind() == SyntaxKind.ThisConstructorInitializer ||
-                nodeToBind.Kind() == SyntaxKind.BaseConstructorInitializer ||
-                nodeToBind.Kind() == SyntaxKind.PrimaryConstructorBaseType || // initializer for a record constructor
-                nodeToBind.Kind() == SyntaxKind.SwitchExpressionArm ||
-                nodeToBind.Kind() == SyntaxKind.ArgumentList && (nodeToBind.Parent is ConstructorInitializerSyntax || nodeToBind.Parent is PrimaryConstructorBaseTypeSyntax) ||
-                nodeToBind.Kind() == SyntaxKind.GotoCaseStatement || // for error recovery
-                nodeToBind.Kind() == SyntaxKind.VariableDeclarator &&
+                nodeToBind.IsKind(SyntaxKind.CasePatternSwitchLabel) ||
+                nodeToBind.IsKind(SyntaxKind.ThisConstructorInitializer) ||
+                nodeToBind.IsKind(SyntaxKind.BaseConstructorInitializer) ||
+                nodeToBind.IsKind(SyntaxKind.PrimaryConstructorBaseType) || // initializer for a record constructor
+                nodeToBind.IsKind(SyntaxKind.SwitchExpressionArm) ||
+                nodeToBind.IsKind(SyntaxKind.ArgumentList) && (nodeToBind.Parent is ConstructorInitializerSyntax || nodeToBind.Parent is PrimaryConstructorBaseTypeSyntax) ||
+                nodeToBind.IsKind(SyntaxKind.GotoCaseStatement) || // for error recovery
+                nodeToBind.IsKind(SyntaxKind.VariableDeclarator) &&
                     new[] { SyntaxKind.LocalDeclarationStatement, SyntaxKind.ForStatement, SyntaxKind.UsingStatement, SyntaxKind.FixedStatement }.
                         Contains(nodeToBind.Ancestors().OfType<StatementSyntax>().First().Kind()) ||
```

Comment:
Out of scope for the analyzer. But this could be `nodeToBind.Ancestors().OfType<StatementSyntax>().First.IsKind(kind1, kind2, kind3, kind4)` if this API is available here, or a series of `||`ed `IsKind(...)` conditions. This also avoid the need of array allocation.